spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/hystrix/HystrixCircuitBreakerConfigFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.hystrix;
+
+import com.netflix.hystrix.HystrixCommandProperties;
+
+/**
+ * Provides configuration for circuit breakers.
+ *
+ * @author Ryan Baxter
+ */
+public interface HystrixCircuitBreakerConfigFactory {
+
+	public HystrixCommandProperties.Setter get(String id);
```

Comment:
Not sure the `id` parameter here makes sense anymore since `HystrixCircuitBreakerConfigFactory` is passed in via the builder.  You could in theory create one factory and use it in all builders in which case it would make sense to keep the `id` parameter.  Thoughts?

Reply:
Yeah, I guess I'm questioning the need for this factory. Like `HystrixCommands` you could just have builder methods for the two `Setter`s.

Reply:
I think its needed to abstract the configuration differences between various circuit breaker libraries.  For example with Hystrix there is only 1 configuration class while with R4J there are two (circuit breaker and timer), the builder interface wouldnt work if we had methods for each configuration class required

Reply:
Except this doesn't extend anything and is specific to hystrix.

Reply:
Yes but the builder does

Reply:
Just to make things clearer....

Different circuit breaker implementation may have different ways to configure things.  For example with Hystrix you can configure the circuit breaker in the timeout in on object, `HystrixCommandProperties`.  However in R4J circuit breaker and timeouts are two distinct pieces of functionality and have their own config objects (and their own modules for that matter).  So to configure the R4J implementation we would need two config objects, `TimeLimiterConfig`, and `CircuitBreakerConfig`.  Since we want `CircuitBreakerBuilder` to be a common interface which passes the "config" for the circuit breaker we would need something to encapsulate the configuration for the circuit breaker so that the `CircuitBreakerBuilder` interface can be the same across all implementations.  

For reference here is the R4J config factory https://github.com/spring-cloud-incubator/spring-cloud-r4j/blob/master/spring-cloud-r4j-core/src/main/java/org/springframework/cloud/r4j/core/R4JConfigFactory.java

Now an argument can be made to remove the `id` parameter as I mentioned above, but we could also make the argument that is makes sense to keep it as well https://github.com/spring-cloud/spring-cloud-netflix/pull/3277#discussion_r234062774